### ISSUE TITLE
pod install incompatible with use_frameworks!

### ISSUE THREAD

**Issue by BrendanVadacoraBell** [smartlook/smartlook-react-native-sdk#25]
Hi, coincidently we were working on a new version at the time that should fix this issue. It was released last Friday (October 4th). Please update to the latest version and let us know whether the issue is solved.

_Originally posted by @pavel-kroh-smartlook in https://github.com/smartlook/smartlook-ios-sdk/issues/10#issuecomment-539508680_

I have pods in my project that require `use_frameworks!`. Unfortunately, when trying to pod install and autolink this library, I get the following error message `target has transitive dependencies that include statically linked binaries:`somepath/ios/Pods/Smartlook/Smartlook/Smartlook.framework)`.

According to the referenced post, the author of the issue was able to use `use_frameworks!` in the newer version of smartlook-ios-sdk. However, I see this library is still using a version prior to this fix. Please could the ios-sdk dependency be upgraded to a newer version in hopes that this will allow `pod install` to succeed with `use_frameworks!` in our Podfile.

RN 0.61.5
Cocoapods 1.8.4
salesforce-react-native-wrapper 0.23.0 and 0.24.0 experience this issue.

**Comment by rtman** [comment]
@pavel-kroh-smartlook any update on this? I'm also facing this issue.

**Comment by pavel-kroh-smartlook** [comment]
"_With CocoaPods 1.5.0, developers are no longer restricted into specifying use_frameworks! in their Podfile in order to install pods that use Swift._" http://blog.cocoapods.org/CocoaPods-1.5.0/

So, no `use_frameworks!` anymore.

However, we probably found a way how to solve this _thing_ from our side, too, and the tweak should be released soon.

In the meantime, you can probably fix it locally by editing `node_modules/smartlook-react-native-wrapper/smartlook-react-native-bridge.podspec` by adding the following flag into it `s.static_framework = true`

**Comment by rtman** [comment]
@pavel-kroh-smartlook 

> "With CocoaPods 1.5.0, developers are no longer restricted into specifying use_frameworks! in their Podfile in order to install pods that use Swift." http://blog.cocoapods.org/CocoaPods-1.5.0/
> 
> So, no use_frameworks! anymore.

This isnt up to us, a certain sdk is requiring this so I'm not sure this is a solution.


> In the meantime, you can probably fix it locally by editing node_modules/smartlook-react-native-wrapper/smartlook-react-native-bridge.podspec by adding the following flag into it s.static_framework = true

Have you tested this?

**Comment by pavel-kroh-smartlook** [comment]
Yes, we did test it.

**tl;dr**

Smartlook SDK is a native software library. We cannot guarantee it will work in all random combinations with 3rd partly libraries and toolkits configurations. We make our best effort to help our users to solve such situations.

**Proposed solution**

In this case, some (obsolete) requirement of a 3rd party component is incompatible with the current Smartlook react native wrapper. After some investigation, we found a possible workaround on our side to accommodate it and, in order to speed things up, we also hinted how it could be applied locally before it gets into our standard distribution.

**How was the proposed solution tested.**

1. `npx react-native init` was used to create a clean Hello World React app.

2. It compiled and run as expected.

3. *Without adding Smartlook into it*, `use_frameworks!` was added to its `Podfile`. The subsequent `pod install` failed with this error message

`[!] The 'Pods-SLTest01' target has transitive dependencies that include statically linked binaries: (…/SLTest01/ios/Pods/CocoaLibEvent/lib/libevent.a, …/SLTest01/ios/Pods/CocoaLibEvent/lib/libevent_core.a, …/SLTest01/ios/Pods/CocoaLibEvent/lib/libevent_extra.a, …/SLTest01/ios/Pods/CocoaLibEvent/lib/libevent_pthreads.a, …/SLTest01/ios/Pods/OpenSSL-Universal/ios/lib/libcrypto.a, and …/SLTest01/ios/Pods/OpenSSL-Universal/ios/lib/libssl.a)`

4. Thus, even clean default React Native apps is incompatible with `use_frameworks!`, it breaks some very standard and essential libraries.

5. We went further, and added Smartlook into the project. The subsequent `pod install` failed again, with this error message that includes `Smartlook.framework`:

`[!] The 'Pods-SLTest01' target has transitive dependencies that include statically linked binaries: (…/SLTest01/ios/Pods/CocoaLibEvent/lib/libevent.a, …/SLTest01/ios/Pods/CocoaLibEvent/lib/libevent_core.a, …/SLTest01/ios/Pods/CocoaLibEvent/lib/libevent_extra.a, …/SLTest01/ios/Pods/CocoaLibEvent/lib/libevent_pthreads.a, …/SLTest01/ios/Pods/OpenSSL-Universal/ios/lib/libcrypto.a, …/SLTest01/ios/Pods/OpenSSL-Universal/ios/lib/libssl.a, and …/SLTest01/ios/Pods/Smartlook/Smartlook/Smartlook.framework)`

6. Our suggested solution was applied, i.e., flag `s.static_framework = true` was added to the `node_modules/smartlook-react-native-wrapper/smartlook-react-native-bridge.podspec` 

7. The next `pod install` failed again, this time **without** listing `Smartlook.framework` as one of the causes.

8. In order to test if our solution does not break something else, `use_frameworks!` was removed from the `Podfile`, and Smartlook setup was added to the app.

9. `pod install` worked again and the app compiled, run and Smartlook started with recording the app as expected.

The honest conclusion from this is that our suggested solution probably solves the issue with `use_frameworks!`.

**Comment by rtman** [comment]
@pavel-kroh-smartlook thanks for the explanation!
